src/cpuid/src/amd/mod.rs

```diff
@@ -28,6 +28,24 @@ mod registers;
 #[derive(Debug, Clone, Eq, PartialEq)]
 pub struct AmdCpuid(pub std::collections::BTreeMap<CpuidKey, CpuidEntry>);
 
+impl AmdCpuid {
+    /// Include leaves from `other` that are not present in `self`.
+    #[inline]
+    #[must_use]
+    pub fn include_leaves_from(mut self, other: Self) -> Self {
```

Comment:
Started wondering, can this go to the CpuidTrait? The implementations look very similar.

Reply:
`CpuidTrait` doesn't implement any setters and its also implemented for `RawCpuid` and `kvm_bindings::CpuId`.

Reply:
Well, ```apply_brand_string()``` looks like a setter (updates some leaves).

> its also implemented for RawCpuid and kvm_bindings::CpuId

Not sure I understand, what is implemented?

Reply:
`apply_brand_string()` uses `get_mut()` to get mutable references to exist leaves. There currently isn't a method in `CpuidTrait` to push new leaves.

Reply:
ok, fair enough.